sphinx_gallery/backreferences.py

```diff
@@ -61,47 +61,56 @@ def visit_Attribute(self, node):
             self.visit(node)
 
     def get_mapping(self):
+        imported_names_split = \
+            [key.split('.') for key in self.imported_names.keys()]
+        if imported_names_split:
+            max_import_splits = len(max(imported_names_split, key=len))
+        else:
+            max_import_splits = 1
         for name in self.accessed_names:
-            local_name = name.split('.', 1)[0]
-            remainder = name[len(local_name):]
-            class_attr = False
-            if local_name in self.imported_names:
-                # Join import path to relative path
-                full_name = self.imported_names[local_name] + remainder
-                if local_name in self.global_variables:
+            for i in range(max_import_splits):
```

Comment:
`i` is a difficult variable name to read, maybe better to use `ii` or something more explicit like `split_level` (or even `si` as an abbreviation). If you don't use it then `_` is a standard "will not be used" name.